codalab/worker/runtime/kubernetes_runtime.py

```diff
@@ -0,0 +1,231 @@
+        if configuration.host == "https://codalab-control-plane:6443":
+            # Don't verify SSL if we are connecting to a local cluster for testing / development.
+            configuration.verify_ssl = False
+            configuration.ssl_ca_cert = None
+            del configuration.api_key_prefix['authorization']
+            del configuration.api_key['authorization']
+            configuration.debug = False
+
+        self.k8_client: client.ApiClient = client.ApiClient(configuration)
+        self.k8_api: client.CoreV1Api = client.CoreV1Api(self.k8_client)
+
+    def get_nvidia_devices(self, use_docker=True):
+        """
+        Returns a Dict[index, UUID] of all NVIDIA devices available to docker
+
+        Arguments:
+            use_docker: whether or not to use a docker container to run nvidia-smi.
+
+        Raises docker.errors.ContainerError if GPUs are unreachable,
+            docker.errors.ImageNotFound if the CUDA image cannot be pulled
+            docker.errors.APIError if another server error occurs
+        """
+        return {}
+
+    def get_container_ip(self, network_name: str, pod_name: str):
+        """Returns (finished boolean, exitcode or None of bundle, failure message or None)"""
+        pod = self.k8_api.read_namespaced_pod_status(pod_name, "default")
+        return pod.status.pod_ip
+
+    def start_bundle_container(
+        self,
+        bundle_path,
+        uuid,
+        dependencies,
+        command,
+        docker_image,
+        network=None,
+        cpuset=None,
+        gpuset=None,
+        request_cpus=0,
+        request_gpus=0,
+        memory_bytes=0,
+        detach=True,
+        tty=False,
+        runtime=DEFAULT_RUNTIME,
+        shared_memory_size_gb=1,
+    ) -> str:
+        if not command.endswith(';'):
+            command = '{};'.format(command)
+        # Explicitly specifying "/bin/bash" instead of "bash" for bash shell to avoid the situation when
+        # the program can't find the symbolic link (default is "/bin/bash") of bash in the environment
+        command = ['/bin/bash', '-c', '( %s ) >stdout 2>stderr' % command]
+        working_directory = '/' + uuid
+        container_name = 'codalab-run-%s' % uuid
+        limits = {'cpu': request_cpus, 'memory': memory_bytes}
+        if request_gpus > 0:
+            limits['nvidia.com/gpu'] = request_gpus  # Configure NVIDIA GPUs
+        config: Dict[str, Any] = {
+            'apiVersion': 'v1',
+            'kind': 'Pod',
+            'metadata': {'name': container_name},
+            'spec': {
+                'containers': [
+                    {
+                        'name': container_name,
+                        'image': docker_image,
+                        'command': command,
+                        'workingDir': working_directory,
+                        'env': [
+                            {'name': 'HOME', 'value': working_directory},
+                            {'name': 'CODALAB', 'value': 'true'},
```

Comment:
Nit: I notice this is done in docker_utils as well, but I'm not totally clear on why we need to set this environment variable?

Reply:
It's just something we do in all runs (presumably a bundle could check if CODALAB=true to do some codalab-specific behavior), so adding this to preserve this functionality in case bundles are depending on it.